File: notification_service/src/main/java/via/group1/notification_service/rabbitmq/messages/PackageSentMessage.java
package via.group1.notification_service.rabbitmq.messages;

import lombok.Data;
import lombok.Getter;
import lombok.Setter;

@Setter
@Getter
public class PackageSentMessage extends RabbitMqMessage {

    private String senderEmail;
    private String receiverEmail;
    private String senderName;
    private String receiverName;
    private String TrackingNumber;
    private long senderId;
    private long receiverId;

    public PackageSentMessage() {
        super("PackageSent");
    }
}

File: CSharpShared/CSharpShared/RabbitMqMessages/PackageSentMessage.cs
namespace CSharpShared.RabbitMqMessages;

public class PackageSentMessage : RabbitMqMessage
{
    public const string TypeConst = "PackageSent";
    public string ReceiverName { get; set; }
    public string ReceiverEmail { get; set; }
    public string TrackingNumber { get; set; }
    public string SenderName { get; set; }
    public string SenderEmail { get; set; }
    public long? ReceiverId { get; set; }
    public long? SenderId { get; set; }
    
    public PackageSentMessage(string trackingNumber, string receiverName, string receiverEmail, string senderName,
        string senderEmail, long? senderId, long? receiverId) : base(TypeConst)
    {
        TrackingNumber = trackingNumber;
        ReceiverName = receiverName;
        ReceiverEmail = receiverEmail;
        SenderName = senderName;
        SenderEmail = senderEmail;
        SenderId = senderId;
        ReceiverId = receiverId;
    }
}